File: Novels/packages/5-mobile-ios/StoryReader/Model/Chapter.swift
import Foundation

// Chapter model - Converted from 7-shared ChapterDto
struct Chapter: Codable, Identifiable {
    let id: String
    let storyId: String
    let title: String
    let content: String? // Content may be nil if not downloaded
    let contentPath: String? // Path to encrypted content file
    let index: Int
    let wordCount: Int?
    let isLocked: Bool
    let price: Int? // Price in points
    let createdAt: Date?
    let updatedAt: Date?
    
    enum CodingKeys: String, CodingKey {
        case id
        case storyId = "story_id"
        case title, content
        case contentPath = "content_path"
        case index
        case wordCount = "word_count"
        case isLocked = "is_locked"
        case price
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Chapter.kt
package com.storysphere.storyreader.model

data class Chapter(
    val id: String,
    val storyId: String,
    val title: String,
    val order: Int,
    val content: String? = null, // Content loaded from encrypted file
    val contentFilePath: String? = null, // Path to encrypted content file
    val wordCount: Int = 0,
    val isPaid: Boolean = false,
    val price: Int? = null, // Price in points
    val isUnlocked: Boolean = false,
    val downloadStatus: DownloadStatus = DownloadStatus.NOT_DOWNLOADED,
    val downloadProgress: Int = 0,
    val lastDownloadedAt: Long? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

enum class DownloadStatus {
    NOT_DOWNLOADED,
    QUEUED,
    DOWNLOADING,
    DOWNLOADED,
    FAILED
}

